test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,104 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	virtualservicev1 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	istiosecurityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	telemetryv1alpha1 "istio.io/client-go/pkg/apis/telemetry/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	k8sclient "k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	maistrav1 "maistra.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	clientconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	WorkingNamespace                        = "default"
+	ModelRegistryDeploymentPath             = "./test/data/model-registry/modelregistry_deployment.yaml"
+	ModelRegistryDatabaseDeploymentPath     = "./test/data/model-registry/database_deployment.yaml"
+	ServingRuntimePath1                     = "./test/data/deploy/kserve-openvino-serving-runtime-1.yaml"
+	InferenceServiceWithModelVersionPath    = "./test/data/deploy/inference-service-with-model-version.yaml"
+	InferenceServiceWithoutModelVersionPath = "./test/data/deploy/inference-service-without-model-version.yaml"
+	InferenceServiceWithInfServiceIdPath    = "./test/data/deploy/inference-service-to-delete.yaml"
+	timeout                                 = time.Second * 30
+	interval                                = time.Millisecond * 50
+)
+
+var (
+	scheme = runtime.NewScheme()
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	cli    client.Client
+	kc     *k8sclient.Clientset
+)
+
+// Run e2e tests using the Ginkgo runner.
+// ODH model controller should be already running in the cluster where
+// these tests will run against
+func TestE2E(t *testing.T) {
```

Comment:
After looking at the tests, I think _kind_ to run the E2Es should not be required.

Probably, these E2Es can run with a similar setup as the one described here: https://book.kubebuilder.io/cronjob-tutorial/writing-tests.

It will be great if you can figure out how to run the tests without Kind.

Reply:
Well, IIUC how `envtest` works I cannot startup a real service to interact with (in this case the model registry service) - that's why those tests leverages on an existing cluster up and running (either with `Kind` or any other tool).

Do you have any particular idea on how I can startup and interact with a real service?

Reply:
As we briefly discussed via chat, my only idea is to use podman/docker to start the model registry and declaring the `MLMD_ADDRESS` environment variable to the right value before running the tests. This would simplify the setup to run the E2Es and the tests may run faster in environments like GitHub Actions.

Indeed, the main motivation for suggesting moving away from Kind was because I was thinking that a subsequent PR would create GitHub Actions workflow to run these new tests, but I think the setup depends on the platform that we are going to choose for running them. Also, I'm not sure if moving away from Kind would lead to more complex code in `BeforeEach` and `AfterEach` functions.

As an aside, to me the tests look closer to units than to E2Es, because the tests need to pre-seed the registry. The way I perceive an E2E test is by using the actual components that would interact on the whole flow. So, this was an aside motivation for suggesting a simpler setup. But this may just be my own expectation for E2Es, so this may not be valid :-)

So, this is just feedback. I don't see a mandatory change here :slightly_smiling_face:, nor this should be a blocker for merging. What is important is to have tests, and we have them! We can stay with Kind for this PR and, if needed, we can later change.

Reply:
I'd agree with Edgar's suggestion here. I'd want to stay away from prescribing an environment where the E2E tests should run and it would be good to have e2e tests be platform agnostic/configurable. But also agreed that it can be a future enhancement

Reply:
I agree with you guys, I tried to make them as generic as possible such that they could be run against an existing cluster (either using Kind or minikube or openshift-ci) - obviously I tested it using Kind only but I don't see why it shouldn't work in other "simulated" or "real" clusters as the only precondition is that the `odh-model-controller` must be running in that cluster.

On the other hand if we can find a way to make those tests running in GHA w/o the need to setup a cluster is much better, I definitely agree with this. Therefore I will create a followup ticket/issue for this enhancement!